File: libs/power-log-parser/src/lib/action-parser.ts
import { Node } from './models';
import { StateType } from './state/parser-state';
import { GameEventProvider } from './game-event';

export interface ActionParser {
	readonly ParserName: string;
	AppliesOnNewNode(node: Node, stateType: StateType): boolean;
	AppliesOnCloseNode(node: Node, stateType: StateType): boolean;
	CreateGameEventProviderFromNew(node: Node): GameEventProvider[] | null;
	CreateGameEventProviderFromClose(node: Node): GameEventProvider[] | null;
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Events/ActionParser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using HearthstoneReplays.Parser;

namespace HearthstoneReplays.Events
{
    interface ActionParser
    {
        bool AppliesOnNewNode(Node node, StateType stateType);
        bool AppliesOnCloseNode(Node node, StateType stateType);
        List<GameEventProvider> CreateGameEventProviderFromNew(Node node);
        List<GameEventProvider> CreateGameEventProviderFromClose(Node node);
    }
}
